packages/adapters/bitcoin/src/connectors/SatsConnectConnector.ts

```diff
@@ -134,31 +151,73 @@ export class SatsConnectConnector extends ProviderEventEmitter implements Bitcoi
     return res.txid
   }
 
-  private getWalletProvider() {
+  protected getWalletProvider() {
     return getProviderById(this.wallet.id) as BitcoinProvider
   }
 
-  private async internalRequest<Method extends keyof SatsConnectRequests>(
+  protected async internalRequest<Method extends keyof SatsConnectRequests>(
     method: Method,
     options: Params<Method>
-  ) {
-    const response = await this.getWalletProvider().request(method, options)
+  ): Promise<RpcSuccessResponse<Method>['result']> {
+    const response = await this.getWalletProvider()
+      .request(method, options)
+      .catch(error => {
+        if ('jsonrpc' in error && 'error' in error) {
+          return error as RpcErrorResponse
+        }
+
+        throw error
+      })
 
     if ('result' in response) {
       return response.result
     }
 
-    throw new Error(response.error.message)
+    throw { ...response.error, name: 'RPCError' } as Error
+  }
+
+  private bindEvents() {
+    this.unbindEvents()
+
+    const provider = this.getWalletProvider()
+
+    if (typeof provider.addListener !== 'function') {
+      console.warn('SatsConnectConnector:bindEvents - wallet provider does not support events')
+
```

Comment:
if we are logging this, then I'd add at least the name or id so we instantly know what's off